cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -145,12 +148,15 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
   @DELETE
   @Path("/namespaces/{namespace-id}")
   public void delete(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespace) {
+    if (!conf.getBoolean(Constants.Dangerous.UNRECOVERABLE_RESET,
```

Comment:
We shouldn't need this unrecoverable reset for namespace delete. This should be handled by ACLs when we have authorization.